Sources/SourceKitLSP/Swift/MacroExpansion.swift

```diff
@@ -74,50 +74,38 @@ extension SwiftLanguageService {
-
-        completeExpansionDirectoryName += "\(bufferName)-"
-
-        do {
-          try FileManager.default.createDirectory(
-            at: macroExpansionBufferDirectoryURL,
-            withIntermediateDirectories: true
-          )
-        } catch {
-          throw ResponseError.unknown(
-            "Failed to create directory for macro expansion buffer at path: \(macroExpansionBufferDirectoryURL.path)"
-          )
+        // Custom Scheme for Macro Expansions:
+        // `swift-macro-expansion://[path_to_source_file]/LaCb-LcCd.swift?lowerL=&lowerC=&upperL=&upperC=&bufferName=`
+        //  ---------------------   --------------------- --------------- ------------------------------- -----------
+        //           1                       2                3                   4                     5
+        //
+        // 1. Scheme
+        // 2. The path to the source file
+        // 3. Position range of macro after expansion (used for display purpose in VS Code)
+        // 4. Selection range of the cursor
+        // 5. Buffer Name of the required Macro Expansion
+
+        guard var components = URLComponents(string: sourceFileURL.absoluteString) else {
+          throw ResponseError.unknown("Invalid URL")
         }
-
-        // name of the source file
-        let macroExpansionFileName = sourceFileURL.deletingPathExtension().lastPathComponent
-
-        // github permalink notation for position range
-        let macroExpansionPositionRangeIndicator =
-          "L\(macroEdit.range.lowerBound.line + 1)C\(macroEdit.range.lowerBound.utf16index + 1)-L\(macroEdit.range.upperBound.line + 1)C\(macroEdit.range.upperBound.utf16index + 1)"
-
-        let macroExpansionFilePath =
-          macroExpansionBufferDirectoryURL
-          .appendingPathComponent(
-            "\(macroExpansionFileName)_\(macroExpansionPositionRangeIndicator).\(sourceFileURL.pathExtension)"
-          )
-
-        do {
-          try macroEdit.newText.write(to: macroExpansionFilePath, atomically: true, encoding: .utf8)
-        } catch {
-          throw ResponseError.unknown(
-            "Unable to write macro expansion to file path: \"\(macroExpansionFilePath.path)\""
-          )
+        components.scheme = "swift-macro-expansion"
+        components.path +=
+          "/L\(macroEdit.range.lowerBound.line + 1)C\(macroEdit.range.lowerBound.utf16index + 1)-L\(macroEdit.range.upperBound.line + 1)C\(macroEdit.range.upperBound.utf16index + 1).swift"
+        let queryItems = [
+          URLQueryItem(name: "lowerL", value: String(expandMacroCommand.positionRange.lowerBound.line)),
+          URLQueryItem(name: "lowerC", value: String(expandMacroCommand.positionRange.lowerBound.utf16index)),
+          URLQueryItem(name: "upperL", value: String(expandMacroCommand.positionRange.upperBound.line)),
+          URLQueryItem(name: "upperC", value: String(expandMacroCommand.positionRange.upperBound.utf16index)),
```

Comment:
We don’t really need to optimize for a few characters here, so let’s spell out `L` to `Line` and `C` to `Column`.

Also, I find `from` and `to` more intuitive names than `lower` and `upper`.